numpy/core/include/numpy/ndarrayobject.h

```diff
@@ -284,7 +284,9 @@ PyArray_XDECREF_ERR(PyArrayObject *arr)
 
 
 #ifdef __cplusplus
-}
+#define CONFUSE_EMACS2 }
+CONFUSE_EMACS2
+#undef CONFUSE_EMACS2
```

Comment:
Why is this block shorter than the other one you've mentioned in the same file, which looks like this:

```C
#ifdef __cplusplus
#define CONFUSE_EMACS {
#define CONFUSE_EMACS2 }
extern "C" CONFUSE_EMACS
#undef CONFUSE_EMACS
#undef CONFUSE_EMACS2
/* ... otherwise a semi-smart identer (like emacs) tries to indent
       everything when you're typing */
#endif
```